test/end_to_end_test.py

```diff
@@ -28,43 +30,43 @@ def cleanup_data_dir():
 folder_to_search = base_path + "/Data/input_pdfs"
 cost_sheet_path = base_path + "/Data/blank_cost_sheet.ods"
 invoice_dir = base_path + "/Data/invoices"
-main.run(folder_to_search=folder_to_search,
-             cost_sheet=cost_sheet_path,
-             invoice_dir=invoice_dir,
-             db_regex=main.DEFAULT_DB_REGEX
-             )
 
-try:
-    assert isdir(base_path + "/Data/invoices/ust_2018_06")
+class TestPyDB(unittest.TestCase):
 
-    # check if correct tickets were found and renamed
-    expected_tickets = [u'bahn_ErlMün20180705.pdf',
-                        u'bahn_MünErl20180702.pdf',
-                        u'bahn_MünErl20180710.pdf'
-                        ]
+    def test_e2e_ticket_handling(self):
+        with unittest.mock.patch('getopt.getopt', return_value=([('--config', 'configs/test_config.yml')],[])):
+            main.main()
 
-    assert len(expected_tickets) == len(os.listdir(base_path + "/Data/invoices/ust_2018_06"))
-    assert len(set(expected_tickets).intersection(
-        set([x for x in os.listdir(base_path + "/Data/invoices/ust_2018_06")]))
-    ) == 3
+            try:
+                assert isdir(base_path + "/Data/invoices/ust_2018_06")
 
-    # check if tickets were moved
-    expected_non_tickets = [u"cv.pdf", u"unicodeexample.pdf"]
-    #todo: why do we need this?
-#    assert len(expected_non_tickets) == len(os.listdir(base_path + "/Data/input_pdfs"))
-    assert len(set(expected_non_tickets).intersection(set(os.listdir(base_path + "/Data/input_pdfs")))) == 2
+                # check if correct tickets were found and renamed
+                expected_tickets = [u'bahn_ErlMün20180705.pdf',
```

Comment:
these are only the expected tickets for june 2018, would be nice to also check the other tickets and if the corresponding folder is created correctly